### PR TITLE
[BOT-X] Send new coronavirus reports to Discord

### PR DESCRIPTION
**Pull request by miguelcarreiro**
## Description
This pull request implements coronavirus report notifications to Discord, and `!corona reports` command, which sends all available DGS reports.

Since this PR is urgent, will merge without the approval of at least 1 reviewer.

## Task items:
- [x] Notification of new DGS reports;
- [x] New `!corona reports` command;
- [ ] Jest tests.

## Requirements
Requirements when deploying the current work to an environment (`dev`, `staging` or `production`):

- Add `DGSCORONA_CHANNEL_ID` ENV variable;
- Run Sequelize migrations again (`npx sequelize-cli db:migrate --env [Environment]`).

### CODE REVIEW

File: src/api/Corona.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  const $ = cheerio.load(pageHtml);
+
+  $('.single_content ul li').each((i, elem) => {
+    data.push({
+      link: $(elem).find('a').prop('href'),
+      title: $(elem).text(),
+    });
+  });
+
+  return data;
```
Comment (tiagojpdias) [inline src/api/Corona.js:19]: No need for intermediate variable `data` and pushing into it. `.map()` was built to deal with these cases.

```js
  return $('.single_content ul li').map((i, elem) => ({
    link: $(elem).find('a').prop('href'),
    title: $(elem).text(),
  });
```

File: src/services/Corona.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ *
+ * @async
+ * @param {Object} client
+ */
+const updateReports = async (client) => {
+  const reports = await CoronaApi.getReports();
+
+  const newSearchResults = await Promise.all(reports.map(report => reportNotInDb(report)));
+
+  const newReports = reports.filter(((_report, i) => newSearchResults[i]));
```
Comment (tiagojpdias) [inline src/services/Corona.js:48]: Extra parentheses not needed.

 ```
const newReports = reports.filter((_report, i) => newSearchResults[i]);
```

File: src/services/Discord.js [hunk 3]
```diff
@@ -41,11 +43,16 @@ const splitMessageString = (msgString) => {
 *
 * @param {Object} channel
 * @param {String} message
+* @param {String} fileUrl
 */
-const sendMessageToChannel = (channel, message) => {
+const sendMessageToChannel = (channel, message, fileUrl = undefined) => {
```
Comment (tiagojpdias) [inline src/services/Discord.js:48]: No need explicitly default it to `undefined`. It will implicitly be `undefined` if no variable is provided.

**Review by tiagojpdias — COMMENTED** [review]
Excellent PR, good job.

Despite been already merged I left some remarks about thing which can me improved. Nothing major but nonetheless

**Comment by miguelcarreiro** [comment]
> Excellent PR, good job.
> 
> Despite been already merged I left some remarks about thing which can me improved. Nothing major but nonetheless

Thanks for the feedback! I'll correct these suggestions in the next PR.

File: src/api/Corona.js (hunk 1 above)
Comment (miguelcarreiro) [reply src/api/Corona.js]: Fixed in https://github.com/vostpt/bot/pull/154

File: src/services/Discord.js (hunk 3 above)
Comment (miguelcarreiro) [reply src/services/Discord.js:48]: Fixed in https://github.com/vostpt/bot/pull/154;

File: src/services/Corona.js (hunk 2 above)
Comment (miguelcarreiro) [reply src/services/Corona.js:48]: Fixed in https://github.com/vostpt/bot/pull/154;

**Pull request merged**